src/appleseed.shaders/src/appleseed/as_switch_texture.osl

```diff
@@ -202,7 +203,7 @@ shader as_switch_texture
         grey_id);
 
     int index = (in_cycle_mode == 0)
-        ? (int) mod(hash_id, len)
+        ? remainder(hash_id, len + 1)
```

Comment:
It would be more logical if `len` was really the length of the array, as you did in another shader.

Reply:
I agree it can be clearer, editing now. FYI, the idea is to get a cycling [0,7] range index for texture lookups in one mode, and to get a clamped [0,7] range index in the other mode.